easybuild/tools/systemtools.py

```diff
@@ -133,76 +100,112 @@ def get_core_count():
 
 
 def get_cpu_vendor():
-    """Try to detect the cpu identifier
+    """
+    Try to detect the CPU vendor
 
-    will return INTEL, ARM or AMD constant
+    @return: INTEL, ARM or AMD constant
     """
-    regexp = re.compile(r"^vendor_id\s+:\s*(?P<vendorid>\S+)\s*$", re.M)
-    VENDORS = {
-        'GenuineIntel': INTEL,
-        'AuthenticAMD': AMD,
-    }
+    vendor = None
     os_type = get_os_type()
 
-    if os_type == LINUX:
-        try:
-            txt = read_file('/proc/cpuinfo', log_error=False)
-            arch = UNKNOWN
-            # vendor_id might not be in the /proc/cpuinfo, so this might fail
-            res = regexp.search(txt)
-            if res:
-                arch = res.groupdict().get('vendorid', UNKNOWN)
-            if arch in VENDORS:
-                return VENDORS[arch]
-
-            # some embeded linux on arm behaves differently (e.g. raspbian)
-            regexp = re.compile(r"^Processor\s+:\s*(?P<vendorid>ARM\S+)\s*", re.M)
-            res = regexp.search(txt)
-            if res:
-                arch = res.groupdict().get('vendorid', UNKNOWN)
-            if ARM in arch:
-                return ARM
-        except IOError, err:
-            raise SystemToolsException("An error occured while determining CPU vendor since: %s" % err)
+    if os_type == LINUX and os.path.exists(PROC_CPUINFO_FP):
+        txt = read_file(PROC_CPUINFO_FP)
+        arch = UNKNOWN
+
+        # vendor_id might not be in the /proc/cpuinfo, so this might fail
+        vendor_regex = re.compile(r"^vendor_id\s+:\s*(?P<vendorid>\S+)\s*$", re.M)
+        res = vendor_regex.search(txt)
+        if res:
+            arch = res.group('vendorid')
+        if arch in VENDORS:
+            vendor = VENDORS[arch]
+            tup = (vendor, vendor_regex.pattern, PROC_CPUINFO_FP)
+            _log.debug("Determined CPU vendor on Linux as being '%s' via regex '%s' in %s" % tup)
+
+        # embedded Linux on ARM behaves differently (e.g. Raspbian)
+        vendor_regex = re.compile(r".*:\s*(?P<vendorid>ARM\S+)\s*", re.M)
+        res = vendor_regex.search(txt)
+        if res:
+            vendor = ARM
+            tup = (vendor, vendor_regex.pattern, PROC_CPUINFO_FP)
+            _log.debug("Determined CPU vendor on Linux as being '%s' via regex '%s' in %s" % tup)
 
     elif os_type == DARWIN:
-        out, exitcode = run_cmd("sysctl -n machdep.cpu.vendor")
+        cmd = "sysctl -n machdep.cpu.vendor"
+        out, ec = run_cmd(cmd)
         out = out.strip()
-        if not exitcode and out and out in VENDORS:
-            return VENDORS[out]
+        if ec == 0 and out and out in VENDORS:
+            vendor = VENDORS[out]
+            _log.debug("Determined CPU vendor on DARWIN as being '%s' via cmd '%s" % (vendor, cmd))
+
+    if vendor is None:
+        vendor = UNKNOWN
+        _log.warning("Could not determine CPU vendor on %s, returning %s" % (os_type, vendor))
+
+    return vendor
+
+
+def get_cpu_family():
```

Comment:
`get_cpu_vendor` should return `IBM` when on a Linux/POWER system, I guess

and I don't know why it would be useful, who knows :)